### PR TITLE
Disallow restricted users from placing orders at the osu!store.

### PR DESCRIPTION
**Pull request by peppy**
They like to fuck things up (surprise surprise)

### CODE REVIEW

**Comment by nanaya** [comment]
I have something really similar to this, just the difference is mine simply blocks entire non-`GET` requests instead with this additional check: `!$request->isMethod('GET')` (and just add the middleware without specifying the actions).

**Comment by peppy** [comment]
where?

File: app/Http/Middleware/CheckUserRestricted.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+     * Handle an incoming request.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure                 $next
+     *
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if ($this->auth->check() && $this->auth->user()->isRestricted()) {
```
Comment (nanaya) [inline app/Http/Middleware/CheckUserRestricted.php]: here (the `GET` method check).

Comment (peppy) [inline app/Http/Middleware/CheckUserRestricted.php]: potentially it may be used for get requests too though, right?

**Pull request merged**